File: ios/runanywhere-sdks-latest/sdk/runanywhere-flutter/packages/runanywhere_llamacpp/ios/Classes/LlamaCppPlugin.swift
import Flutter
import UIKit

/// RunAnywhere LlamaCPP Flutter Plugin - iOS Implementation
///
/// This plugin provides the native bridge for the LlamaCPP backend on iOS.
/// The actual LLM functionality is provided by RABackendLLAMACPP.xcframework.
public class LlamaCppPlugin: NSObject, FlutterPlugin {

    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(
            name: "runanywhere_llamacpp",
            binaryMessenger: registrar.messenger()
        )
        let instance = LlamaCppPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "getPlatformVersion":
            result("iOS " + UIDevice.current.systemVersion)
        case "getBackendVersion":
            result("0.1.4")
        case "getBackendName":
            result("LlamaCPP")
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}

File: ios/runanywhere-sdks-latest/sdk/runanywhere-flutter/packages/runanywhere_llamacpp/android/src/main/kotlin/ai/runanywhere/sdk/llamacpp/LlamaCppPlugin.kt
package ai.runanywhere.sdk.llamacpp

import android.os.Build
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/**
 * RunAnywhere LlamaCPP Flutter Plugin - Android Implementation
 *
 * This plugin provides the native bridge for the LlamaCPP backend on Android.
 * The actual LLM functionality is provided by RABackendLlamaCPP native libraries (.so files).
 */
class LlamaCppPlugin : FlutterPlugin, MethodCallHandler {
    private lateinit var channel: MethodChannel

    companion object {
        private const val CHANNEL_NAME = "runanywhere_llamacpp"
        private const val BACKEND_VERSION = "0.1.4"
        private const val BACKEND_NAME = "LlamaCPP"

        init {
            // Load LlamaCPP backend native libraries
            try {
                System.loadLibrary("rac_backend_llamacpp_jni")
            } catch (e: UnsatisfiedLinkError) {
                // Library may not be available in all configurations
                android.util.Log.w("LlamaCpp", "Failed to load rac_backend_llamacpp_jni: ${e.message}")
            }
        }
    }

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, CHANNEL_NAME)
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        when (call.method) {
            "getPlatformVersion" -> {
                result.success("Android ${Build.VERSION.RELEASE}")
            }
            "getBackendVersion" -> {
                result.success(BACKEND_VERSION)
            }
            "getBackendName" -> {
                result.success(BACKEND_NAME)
            }
            else -> {
                result.notImplemented()
            }
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}
